src/include/duckdb/storage/compression/alprd/algorithm/alprd.hpp

```diff
@@ -0,0 +1,221 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/storage/compression/alprd/algorithm/alprd.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/storage/compression/patas/patas.hpp"
+#include "duckdb/storage/compression/alprd/alprd_constants.hpp"
+
+#include "duckdb/common/bitpacking.hpp"
+#include "duckdb/common/common.hpp"
+#include "duckdb/common/pair.hpp"
+#include "duckdb/common/unordered_map.hpp"
+#include "duckdb/common/limits.hpp"
+
+namespace duckdb {
+
+namespace alp {
+
+template <class T, bool EMPTY>
+class AlpRDCompressionState {
+public:
+	using EXACT_TYPE = typename FloatingToExact<T>::type;
+
+	AlpRDCompressionState() : right_bw(0), left_bw(0), exceptions_count(0) {
+	}
+
+	void Reset() {
+		left_bp_size = 0;
+		right_bp_size = 0;
+		exceptions_count = 0;
+	}
+
+public:
+	uint8_t right_bw;
```

Comment:
These abbreviations are not immediately clear to me, can we please not abbreviate this or at least put comments above these that help clear this up?
for example:
`//! Right bitwidth`

Reply:
True. We have changed the variable names to be more representative